src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/KestrelThread.cs

```diff
@@ -66,6 +66,11 @@ public KestrelThread(KestrelEngine engine)
 
         public Task StartAsync()
         {
+#if !DEBUG
+            // Mark the thread as being as unimportant to keeping the process alive.
+            // Don't do this for debug builds, so we know if the thread isn't terminating.
+            _thread.IsBackground = true;
+#endif
```

Comment:
I verified this works by moving libuv.dll so it couldn't be loaded, but I think this makes more sense in the ctor after we set `_thread.Name`.